### PR TITLE
MVCC read-set with boundaries

### PR DESCRIPTION
**Pull request by jeroiraz**
This PR sets a configurable maximum size for the read-set used to validate transaction conflicts

Note: This PR may be reviewed after PR #1467

### CODE REVIEW

File: embedded/store/immustore.go [hunk 1]
```diff
@@ -158,6 +159,7 @@ type ImmuStore struct {
 	synced                bool
 	syncFrequency         time.Duration
 	maxActiveTransactions int
+	mvccReadSetLimit      int
```
Comment (tomekkolo) [inline embedded/store/immustore.go:162]: would be great to see comments explaining those fields :P

File: embedded/store/options.go [hunk 2]
```diff
@@ -73,6 +74,7 @@ type Options struct {
 	CompactionDisabled bool
 
 	MaxActiveTransactions int
+	MVCCReadSetLimit      int
```
Comment (tomekkolo) [inline embedded/store/options.go:77]: please add comment what this limit means

File: embedded/store/ongoing_tx.go [hunk 3]
```diff
@@ -273,7 +274,12 @@ func (tx *OngoingTx) GetWithFilters(key []byte, filters ...FilterFn) (ValueRef,
 			filters: filters,
 		}
 
+		if tx.readsetSize == tx.st.mvccReadSetLimit {
```
Comment (arriqaaq) [inline embedded/store/ongoing_tx.go:277]: you could use a method on the tx here, something like `isMVCCLimitReached` or something similar, as this check is repeated a lot

**Review by arriqaaq — DISMISSED** [review]
lgtm, minor comment on refactor

**Pull request merged**